db/database.go

```diff
@@ -194,9 +195,12 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
 	// Initialize the tap Listener for notify handling
 	context.tapListener.Init(bucket.GetName())
 
-	// If not using channel index, start the tap feed
-	if options.IndexOptions == nil {
-		if err = context.tapListener.Start(bucket, options.TrackDocs, func(bucket string, err error) {
+	// If not using channel index or using shadowing, start the tap feed
+	if options.IndexOptions == nil || options.Shadowing {
```

Comment:
I've rebased this change against master and created a new PR which just adds a `Shadowing` property to the `DatabaseContextOptions` struct.

see #2213